### ISSUE TITLE
"Method not found: 'Void Lib.Net.Http.WebPush.PushServiceClient.set_MaxRetriesAfter(Int32)

### ISSUE THREAD

**Issue by knopa** [tpeczek/Lib.Net.Http.WebPush#28]
When call **_pushClient.RequestPushMessageDeliveryAsync**
Got exception 
```
Method not found: 'Void Lib.Net.Http.WebPush.PushServiceClient.set_MaxRetriesAfter(Int32)'.
System.MissingMethodException: Method not found: 'Void Lib.Net.Http.WebPush.PushServiceClient.set_MaxRetriesAfter(Int32)'.
```

**Comment by tpeczek** [comment]
Hi @knopa,

Now this is an interesting issue, as one does not simply get `MissingMethodException`. Usually that involves either reflection or mismatched assemblies.

I also suspect that this in fact doesn't come from call to `RequestPushMessageDeliveryAsync` as it doesn't attempt to set `MaxRetriesAfter` anywhere in its flow.

If you are not attempting to set `MaxRetriesAfter` in your code in an unusual way, my best guess (and this is purely a guess) is that you are using `Lib.AspNetCore.WebPush` and its dependency injection integration. The dependency injection integration is setting `MaxRetriesAfter` as part of the implementation factory. So this is something which could cause such an exception at runtime when dependency is being resolved. But, for this to happen, it would require an older version of `Lib.Net.Http.WebPush` (one which doesn't have `MaxRetriesAfter` property yet) being currently loaded instead of one which is compatible with `Lib.AspNetCore.WebPush`.

This is probably as much as I can suggest without some kind of repro project.

**Comment by knopa** [comment]
Hi @tpeczek 

You are right - it's happens when trying to resolve `PushServiceClient `as DI

![image](https://user-images.githubusercontent.com/1783681/221021027-c5db5128-428b-4266-90e2-6afd41d8f5cf.png)

I have registered it  and inject it into scoped service.
         ```
services.AddPushServiceClient(options =>
            {
                options.Subject = "https://localhost:5001/";
                options.PublicKey = "BPYSp-";
                options.PrivateKey = "ooDZ";
            });
```

**Comment by knopa** [comment]
I see now, it's not possible to use `Lib.AspNetCore.WebPush` without installing `Lib.Net.Http.WebPush`

I think you need bump version as min dependecy

**Comment by tpeczek** [comment]
You are correct, something got messed up at dependency level.

The 2.1.0 version of  `Lib.AspNetCore.WebPush` has correct minimal dependency (3.1.0):

![image](https://user-images.githubusercontent.com/1446301/221025589-94eab322-3d59-445c-9559-fd69eed5c991.png)

But 2.2.0 version of `Lib.AspNetCore.WebPush` somehow has incorrect minimal dependency (2.2.0):

![image](https://user-images.githubusercontent.com/1446301/221025894-c7cc6269-dbc3-4c8c-90f9-b61a373349ed.png)

So this will have to be fixed and released as 2.2.1. It will probably be the best to unlist 2.2.0 as well.

Thank you for reporting this.
